### DISCUSSION TITLE
v0.19.877

### DISCUSSION THREAD

**Discussion by amandel** [Friends-of-OpenBikeSensor/OpenBikeSensorFirmware#353] [Announcements]
![GitHub downloads](https://img.shields.io/github/downloads-pre/openbikesensor/OpenBikeSensorFirmware/v0.19.877/total) ![GitHub commits since release](https://img.shields.io/github/commits-since/openbikesensor/OpenBikeSensorFirmware/f5bca8cc7489f268352d1bc2f3a0ab2154f7171a?label=commits%20since%20v0.19.877)

Version v0.19.877 based on refs/heads/main f5bca8cc7489f268352d1bc2f3a0ab2154f7171a

## :tada: Major features and improvements

- [b7e83f2](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/b7e83f2fa66df1db5c7401254a66b57bab9e9ab3) - Allow multiple WiFi configurations for different networks (#343)

## :rocket: New features and improvements 

- [8a9fb45](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/8a9fb45052918fd5a4c457ea273aec89ffa05b49) - OpenBikeSensorFirmware-334 Flush CSV/SD buffer when privacy zone is entered (#341)

## :package: Dependency updates

- [e80c0ad](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/e80c0adcee67a2530aba109f861927db27780060) - Chore: update SQ scanner version
- [3f938c3](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/3f938c33693b465bee5568ecabb4faf80c9803fa) - Bump metcalfc/changelog-generator from 4.1.0 to 4.2.0 (#346)
- [d53fea6](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/d53fea65e55413add4449b38cd327d7497d8924d) - platform update to 5.4.0
- [c218640](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/c2186408fc9e2d2d3d52eeb855053b25e00a44f3) - Update SonarQube dependencies (#342)
- [43a7299](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/43a729926ecd52d722e1c11522c6adc9e534023a) - Update library dependencies
- [d2b1bb6](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/d2b1bb6716fbe01339a3114241f833797fa8e642) - Bump metcalfc/changelog-generator from 4.0.1 to 4.1.0 (#337)

## :construction_worker: Changes for developers / internal

- [f5bca8c](http://github.com/openbikesensor/OpenBikeSensorFirmware/commit/f5bca8cc7489f268352d1bc2f3a0ab2154f7171a) - Preparing 0.19 release

<hr /><em>This discussion was created from the release <a href='https://github.com/openbikesensor/OpenBikeSensorFirmware/releases/tag/v0.19.877'>v0.19.877</a>.</em>

**Comment by gluap** [comment]
@amandel On one device that I had pre-flashed with "pio run -t upload" running 0.18-dev I wasn't able to automatically update

- Firmware correctly found the latest version number, but wasn't able to flash 0.19.xx from the weg
- The same device also wasn't able to flash 0.18.xx from the web.
- ``pio run -t erase; pio run -t upload didn't help``
- uploading the latest ``firmware.bin`` didn't help (showed "updating" but came back up with v0.18-dev)
- Using a new SD fixed the issue.
  - Strangely though the original sd works just fine for recording, only gives trouble when flashing. 

So everything is alrigtht :)

Log during failing flash attempt (note to self: the light blue OBS on the cargobike):

```
␛[0;32mI (233) flash-sd: OpenBikeSensorFlash version v0.2.38.␛[0m
␛[0;32mI (243) flash-sd: Reset reason 0x3.␛[0m
␛[0;32mI (243) flash-sd: Running partition: flashApp␛[0m
␛[0;32mI (253) flash-sd: App 'OpenBikeSensorFlash', Version: 'v0.2.38'␛[0m
␛[0;32mI (253) flash-sd: IDF-Version: '4.3.1'␛[0m
␛[0;32mI (263) flash-sd: sha-256: 5fd778b4903689a73b328fe0a161cf4a7b080714b4512308f0c053c9a690fbb7␛[0m
␛[0;32mI (273) flash-sd: date: 'Nov 27 2021', time: '15:01:34'␛[0m
␛[0;32mI (273) flash-sd: Initializing SD card␛[0m
␛[0;32mI (283) gpio: GPIO[5]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 ␛[0m
␛[0;32mI (293) sdspi_transaction: cmd=52, R1 response: command CRC error␛[0m
␛[0;31mE (293) sdmmc_io: sdmmc_io_reset: unexpected return: 0x109␛[0m
␛[0;31mE (303) vfs_fat_sdmmc: sdmmc_card_init failed (0x109).␛[0m
␛[0;32mI (303) gpio: GPIO[5]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 ␛[0m

abort() was called at PC 0x400d1351 on core 0

Backtrace:0x40085422:0x3ffb55e0 0x40085b4d:0x3ffb5600 0x4008a0a6:0x3ffb5620 0x400d1351:0x3ffb5690 0x400f023e:0x3ffb5760
```

**Reply by amandel** [reply]
Hmm, I'm not sure. A broken SD card could definitely be a possible cause. But you write that the SD worked fine otherwise.
Still the flasher app uses a bit different technology and likely a different library to talk to the SD card. 
I'm guessing here, glad that it works now and we hopefully do not see this often.
